utility/src/test/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecoratorTest.kt

```diff
@@ -0,0 +1,76 @@
+package org.oppia.android.util.profile
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.util.profile.CurrentUserProfileIdIntentDecorator.decorateWithUserProfileId
+import org.oppia.android.util.profile.CurrentUserProfileIdIntentDecorator.extractCurrentUserProfileId
+import org.robolectric.annotation.LooperMode
+
+/** Tests for [CurrentUserProfileIdIntentDecorator]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class CurrentUserProfileIdIntentDecoratorTest {
+
+  @Test
+  fun testDecorator_decorateWithProfileId_returnsIntentWithCorrectProfileId() {
+    val intent = Intent().apply {
+      decorateWithUserProfileId(
+        ProfileId.newBuilder().apply {
+          internalId = 0
+        }.build()
+      )
+    }
+    val currentProfileId = intent.extractCurrentUserProfileId()
+    assertThat(currentProfileId.internalId).isEqualTo(0)
+  }
+
+  @Test
+  fun testDecorator_withoutProfileId_returnsIntentWithDefaultProfileId() {
+    val currentProfileId = Intent().extractCurrentUserProfileId()
+    assertThat(currentProfileId).isEqualTo(ProfileId.getDefaultInstance())
+  }
+
+  @Test
+  fun testDecorateIntentWithUserProfileId_emptyProfileId_returnsDefaultProfileIdInstance() {
+    val profileId = ProfileId.newBuilder().apply {}.build()
+
+    val extractedProfileId = Intent().apply {
+      decorateWithUserProfileId(profileId)
+    }.extractCurrentUserProfileId()
+    assertThat(extractedProfileId).isEqualTo(ProfileId.getDefaultInstance())
+  }
+
+  @Test
+  fun testDecorator_decorateBundleWithProfileId_returnsBundleWithCorrectProfileId() {
+    val profileId = ProfileId.newBuilder().apply { internalId = 1 }.build()
+    val bundle = Bundle().apply {
+      decorateWithUserProfileId(profileId)
+    }
+
+    val currentUserProfileId = bundle.extractCurrentUserProfileId()
+    assertThat(currentUserProfileId).isEqualTo(profileId)
+  }
+
+  @Test
+  fun testDecorator_withoutProfileId_returnsBundleWithDefaultProfileId() {
+    val currentProfileId = Bundle().extractCurrentUserProfileId()
+    assertThat(currentProfileId).isEqualTo(ProfileId.getDefaultInstance())
+  }
+
+  @Test
+  fun testDecorateBundleWithUserProfileId_emptyProfileId_returnsDefaultProfileIdInstance() {
+    val profileId = ProfileId.newBuilder().apply {}.build()
```

Comment:
Here & above: you can use ``getDefaultInstance`` for an "empty" proto.

Reply:
Updated.